src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -231,6 +232,19 @@ else if (shouldStartLine && !isOnStartOfLine(details, targetSourceLine)) {
     }
 
     /**
+     * Checks that a right curly should be alone on line or alone with SEMI token is LAMBDA.
+     * @param bracePolicy option for placing the right curly brace
+     * @param details Details for validation
+     * @return true if a right curly should be alone on line or with semi if token is LAMBDA.
+     */
+    private static boolean shouldBeAlone(RightCurlyOption bracePolicy, Details details) {
```

Comment:
@MaksimP 
shouldBeAloneOnLine can be split into two methods: shouldBeAloneOnLine and shouldBeAloneOrSingleLine as now method's name is not correct since it performs checking for to brace policies: ALONE and ALONE_OR_SINGLELINE.